### PR TITLE
Add serialization reference test & bump Legolas compat

### PR DESCRIPTION
**Pull request by ericphanson**
This should be merged before #21!

### CODE REVIEW

File: examples/digits.jl [hunk 1]
```diff
@@ -138,3 +138,23 @@ testmode!(m2)
 output2 = m2(input)
 
 @test output ≈ output2
+
+# Create row:
```
Comment (ericphanson) [inline examples/digits.jl:142]:
```suggestion
```

File: examples/digits.jl [hunk 2]
```diff
@@ -8,13 +8,13 @@ using StableRNGs
...
 using Legolas, LegolasFlux
+using Tables
 
 # This should store all the information needed
 # to construct the model.
-Base.@kwdef struct DigitsConfig
-    seed::Int = 5
-    dropout_rate::Float32 = 0f1
-end
+const DigitsConfig = Legolas.@row("digits-config@1",
```
Comment (ericphanson) [inline examples/digits.jl:15]: Changed to a Legolas object so it serializes easier (otherwise need manual Arrow stuff)

Comment (kleinschmidt) [reply examples/digits.jl:15]: Hm, does legolas pre 0.4 support this?  Looks like it's deserializing as a NamedTuple.

File: examples/digits.jl [hunk 3]
```diff
@@ -58,8 +58,8 @@ end
 # We add our `DigitsConfig` object, as well as the epoch and accuracy.
 const DigitsRow = Legolas.@row("digits.model@1" > "legolas-flux.model@1",
                                config::DigitsConfig,
```
Comment (kleinschmidt) [inline examples/digits.jl:60]: maybe ?

```suggestion
                               config::DigitsConfig = DigitsConfig(config),
```

**Comment by ericphanson** [comment]
Maybe we can drop pre-Legolas v0.4 in this PR; I think that's still non-breaking, nothing from our API changed. (The problem is nested serialization doesn't work right on pre-v0.4, as I'm sure you remember Dave 😄).

**Review by kleinschmidt — APPROVED** [review]
Looks good, I'm fine dropping earlier compat  for the sake of keeping this moving.

**Pull request merged**